shell/platform/android/io/flutter/embedding/FlutterActivity.java

```diff
@@ -0,0 +1,236 @@
+   * {@code FrameLayout} will hold a {@code FlutterFragment}, which displays a {@code FlutterView}.
+   */
+  private void createFlutterFragmentContainer() {
+    FrameLayout container = new FrameLayout(this);
+    container.setId(CONTAINER_ID);
+    container.setLayoutParams(new ViewGroup.LayoutParams(
+        ViewGroup.LayoutParams.MATCH_PARENT,
+        ViewGroup.LayoutParams.MATCH_PARENT
+    ));
+    setContentView(container);
+  }
+
+  /**
+   * If no {@code FlutterFragment} exists in this {@code FlutterActivity}, then a {@code FlutterFragment}
+   * is created and added. If a {@code FlutterFragment} does exist in this {@code FlutterActivity}, then
+   * a reference to that {@code FlutterFragment} is retained in {@code flutterFragment}.
+   */
+  private void createFlutterFragment() {
+    FragmentManager fragmentManager = getFragmentManager();
+
+    flutterFragment = (FlutterFragment) fragmentManager.findFragmentByTag(TAG_FLUTTER_FRAGMENT);
+
+    if (flutterFragment == null) {
+      // No FlutterFragment exists yet. This must be the initial Activity creation. We will create
+      // and add a new FlutterFragment to this Activity.
+      flutterFragment = FlutterFragment.newInstance(
+          isSplashScreenDesired(),
+          getInitialRoute(),
+          getAppBundlePath()
+      );
+      fragmentManager
+          .beginTransaction()
+          .add(CONTAINER_ID, flutterFragment, TAG_FLUTTER_FRAGMENT)
+          .commit();
+    }
+  }
+
+  /**
+   * Sets the color of this {@code Activity}'s status bar to a translucent black, i.e., a dark
+   * overlay.
+   *
+   * Beginning with Android Lollipop, the status bar at the top of the screen can have its color
+   * set at runtime.
+   */
+  @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+  private void configureStatusBarColor() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      Window window = getWindow();
+      window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+      window.setStatusBarColor(0x40000000);
+      window.getDecorView().setSystemUiVisibility(PlatformPlugin.DEFAULT_SYSTEM_UI);
+    }
+  }
+
+  /**
+   * Should the {@code Activity}'s {@code windowBackground} be used as a launch screen
+   * until the first frame of Flutter is rendered?
+   *
+   * This preference is controlled by setting a {@code <meta-data>} tag in the Android
+   * manifest for this {@code Activity}.
+   *
+   * TODO(mattcarroll): Should this preference be set by Intent parameter instead? How would a dev use this in production?
+   */
+  private boolean isSplashScreenDesired() {
+    try {
+      @SuppressLint("WrongConstant")
+      ActivityInfo activityInfo = getPackageManager().getActivityInfo(
+          getComponentName(),
+          PackageManager.GET_META_DATA|PackageManager.GET_ACTIVITIES
+      );
+      Bundle metadata = activityInfo.metaData;
+      return metadata != null && metadata.getBoolean(SPLASH_SCREEN_META_DATA_KEY);
```

Comment:
we should probably also check isTaskRoot or something. When we added this, we were definitely not thinking of the multi-embedding case

Reply:
What's the issue here?

Reply:
Never mind, double nested activities won't end up having the metadata set in the manifest. I forgot how this stuff worked.